### ISSUE TITLE
Cop idea: UselessBlock

### ISSUE THREAD

**Issue by pirj** [rubocop-hq/rubocop#7798]
```ruby
[1,2,3].first { |x| x > 1 }
```

There are quite a lot of methods that do not actually accept a block, but intuitively look like they can optionally accept a block.

Since Ruby core/stdlib methods raise no errors and don't issue any warnings for such misuse, it's a job for ![image](https://user-images.githubusercontent.com/6916/76574883-a0151c00-64ce-11ea-9aca-d248fd768823.png)

## Describe the solution you'd like

Write a cop that detects blocks passed to methods in core/stdlib that don't accept blocks.
Both `method { ... }` and `method(..., &block)` should be flagged.

Methods I have in mind (this list is far from being complete):
 - Array#uniq
 - Array#first
 - ...

### Conditionally optional block

There are cases when a method does accept a block, but only in some circumstances:
```ruby
{a:1}.fetch(:b, 3) { 2 } # => 2
  warning: block supersedes default value argument
```
Does Ruby warn in all cases, in all supported target Ruby versions?

### API changes across Ruby versions

Were there any methods that didn't accept a block in one Ruby version, but do accept a block in a different Ruby version?

### 2.7 delegation

How to deal with 2.7's `...` delegation?
```ruby
def a(...)
  [1,2,3].map(...)
end
a { |x| x*2 } # => [2, 4, 6]
```
It does pass blocks around (even though it probably shouldn't), but it's impossible to detect if the delegating code accepted a block itself.

**Comment by Drenmi** [comment]
Nice. I like the idea. 👍

**Comment by koic** [comment]
The first of all, this is an interesting cop 👍The following is a story that takes the future a little further.
This issue may be solved by the type checker introduced in Ruby 3.0.

> Array#uniq

(On another note) `Array#uniq` accepts a block argument.
https://www.rubydoc.info/stdlib/core/Array:uniq

Well, there are Steep and Sorbet as Ruby type checkers.
The following is an example using Steep.

```console
% gem i steep
% cat Steepfile
target :app do
  check 'lib'
end
% cat lib/example.rb
[1, 2, 2, '2'].first               #=> 1
[1, 2, 2].first {}                 #=> 1 (but the block argument is meaningless)
[1, 2, 2, '2'].uniq                #=> [1, 2, "2"]
[1, 2, 2, '2'].uniq { |n| n.to_s } #=> [1, 2]
```

```console% steep --version0.14.0
% steep check lib/example.rb
lib/example.rb:2:0: UnresolvedOverloading: receiver=::Array[::Integer], method_name=first, method_types=() -> (::Integer | nil) | (::int) -> ::Array[::Integer] ([1, 2, 2].first {})
```

An type error occurs for `[1, 2, 2].first {}`.

So, If an deny list of block argument is created for this cop, some work may be duplicated with the typing for builtin libraries working on https://github.com/ruby/ruby-signature.
